pkg/perf/cache.go

```diff
@@ -0,0 +1,49 @@
+package perf
+
+import (
+	"context"
+	"encoding/json"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+// TestResultData the result test schema
+type TestResultData struct {
+	TestName   string
+	TestNumber uint64
+	Result     interface{}
+}
+
+// GetRequest the get request struct
+type GetRequest struct {
+	TestName   string `json:"test_name"`
+	TestNumber uint64 `json:"test_number"`
+}
+
+// CacheResult set result in redis
+func (pm *PerformanceMonitor) CacheResult(ctx context.Context, resultKey string, resultData TestResultData) error {
+	data, err := json.Marshal(resultData)
+	if err != nil {
+		return errors.Wrap(err, "Error marshaling data to JSON")
+	}
+
+	return pm.RedisClient.Set(resultKey, data, 10*time.Second).Err()
```

Comment:
extract the expiration into an unexported constant `resultTTL`, also 10 is too low imo,

Reply:
Was gonna say that data should never expire, but instead have the `timestamp` of when the test was run. but never expire. Some tests run every 6 hours our once per day. Hence expiration is not needed. so `0` is the right value